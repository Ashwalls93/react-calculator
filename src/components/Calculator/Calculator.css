.calculator-container {
  height: var(--calculator-height);
  width: var(--calculator-width);
  transition: var(--default-transition);
}

.calculator-theme-default {
  background-color: var(--calculator-background-color-default);
}

.calculator-theme-pony {
  background-color: var(--calculator-background-color-pony);
  border-radius: 8px;
}

.calculator-theme-dondi {
  background: url('../../assets/dondi.jpg');
  border-radius: 8px;
}