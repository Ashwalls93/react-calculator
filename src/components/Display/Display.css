.display-container {
  align-items: center;
  display: flex;
  height: var(--display-height);
  padding: 0 4%;
  transition: var(--default-transition);
  width: var(--display-width);
}

.display-theme-default {
  background-color: var(--display-background-color-default);
}

.display-theme-pony {
  background-color: var(--display-background-color-pony);
  border-radius: 8px 8px 0 0;
}

.display-theme-dondi {
  background-color: var(--display-background-color-dondi);
  border-radius: 8px 8px 0 0;
}

.display-value {
  font-size: var(--display-text-size);
  font-family: var(--main-font);
  font-weight: var(--display-text-weight);
  margin-left: auto;
  overflow: hidden;
}

.display-default {
  color: var(--display-text-color-default);
}

.display-pony {
  color: var(--display-text-color-pony);
}

.display-dondi {
  color: var(--display-text-color-dondi);
}