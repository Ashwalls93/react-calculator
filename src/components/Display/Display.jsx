import React from 'react';

const Display = () => {
  return (
    <div className="display-container">
    </div>
  );
}

export default Display;
