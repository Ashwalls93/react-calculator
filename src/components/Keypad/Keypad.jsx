import React from 'react';

const Keypad = () => {
  return (
    <div className="keypad-container">
    </div>
  );
}

export default Keypad;
