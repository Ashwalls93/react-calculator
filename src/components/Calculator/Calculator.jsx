import React, {Component} from 'react';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

class Calculator extends Component {
  constructor(props) {
    super(props);

    this.state = {
      displayValue: '0',
      numbers: ['9', '8', '7', '6', '5', '4', '3', '2', '1', '.', '0', 'ce'],
      operators: ['/', 'x', '-', '+'],
      selectedOperator: '',
      storedValue: '',
    }

    this.callOperator = this.callOperator.bind(this);
    this.setOperator = this.setOperator.bind(this);
    this.updateDisplay = this.updateDisplay.bind(this);
  }

  callOperator = () => {
    let {displayValue, selectedOperator, storedValue} = this.state;
    const updateStoredValue = displayValue;
    
    displayValue = parseInt(displayValue, 10);
    storedValue = parseInt(storedValue, 10);

    switch(selectedOperator) {
      case '+':
        displayValue = storedValue + displayValue;
        break;
      case '-':
        displayValue = storedValue - displayValue;
        break;
      case 'x':
        displayValue = storedValue * displayValue;
        break;
      case '/':
        displayValue = storedValue / displayValue;
        break;
      default:
        displayValue = '0';
    }

    displayValue = displayValue.toString();
    selectedOperator = '';
    if (displayValue === 'NaN' || displayValue === 'Infinity') displayValue ='0';
    this.setState({displayValue, selectedOperator, storedValue: updateStoredValue});
  }

  setOperator = (value) => {
    let {displayValue, selectedOperator, storedValue} = this.state;

    if (selectedOperator === '') {
      storedValue = displayValue;
      displayValue = '0';
      selectedOperator = value;  
    } else {
      selectedOperator = value;
    }

    this.setState({displayValue, selectedOperator, storedValue});
  }

  updateDisplay = (value) => {
    let {displayValue} = this.state;
    
    // prevent multiple occurences of '.'
    if (value === '.' && displayValue.includes('.')) value = '';
    
    if (value === 'ce') {
      // deletes last char in displayValue
      displayValue = displayValue.substr(0, displayValue.length - 1);
      // set displayValue to '0' if displayValue is empty string
      if (displayValue === '') displayValue = '0';
    } else {
      // replace displayValue with value if displayValue equal to '0'
      // else concatenate displayValue and value
      displayValue === '0' ? displayValue = value : displayValue += value;
    }

    this.setState({displayValue});
  }
  
  render () {
    return (
      <div className="calculator-container">
        <Display displayValue="" />
        <Keypad 
          numbers={[]}
          operators={[]}
        />
      </div>
    );
  }
}

export default Calculator;