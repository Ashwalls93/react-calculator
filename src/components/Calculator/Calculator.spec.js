import React from 'react';
import {shallow} from 'enzyme';
import Calculator from './Calculator';
import Display from '../Display/Display';
import Keypad from '../Keypad/Keypad';

describe('Calculator', () => {
  it('should render a <div />', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('should render the Display and Keypad Components', () => {
    const wrapper = shallow(<Calculator />);
    expect(wrapper.containsAllMatchingElements([
      <Display />,
      <Keypad />
    ])).toEqual(true);
  });
});