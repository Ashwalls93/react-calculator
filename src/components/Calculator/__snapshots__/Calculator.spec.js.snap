// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calculator />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Display
          displayValue=""
/>,
        <Keypad
          numbers={Array []}
          operators={Array []}
/>,
      ],
      "className": "calculator-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "displayValue": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "numbers": Array [],
          "operators": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Display
            displayValue=""
/>,
          <Keypad
            numbers={Array []}
            operators={Array []}
/>,
        ],
        "className": "calculator-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "displayValue": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "numbers": Array [],
            "operators": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
